agent/agent.go

```diff
@@ -507,83 +505,12 @@ func (a *Agent) Setup() error {
 	return nil
 }
 
-// WriteSummary writes a summary report that includes the products and op statuses present in the agent's
-// ManifestOps. The intended use case is to write to output at the end of the Agent's Run.
-func (a *Agent) WriteSummary(writer io.Writer) error {
-	t := tabwriter.NewWriter(writer, 0, 0, 2, ' ', 0)
-	headers := []string{
-		"product",
-		string(op.Success),
-		string(op.Fail),
-		string(op.Skip),
-		string(op.Unknown),
-		"total",
-	}
-
-	_, err := fmt.Fprint(t, formatReportLine(headers...))
-	if err != nil {
-		return err
-	}
-
-	// For deterministic output, we sort the products in alphabetical order. Otherwise, ranging over the map
-	// a.ManifestOps directly, we wouldn't know for certain which order the keys - and therefore the rows - would be in.
-	var products []string
-	for k := range a.ManifestOps {
-		products = append(products, k)
-	}
-	sort.Strings(products)
-
-	for _, prod := range products {
-		var success, fail, skip, unknown int
-		ops := a.ManifestOps[prod]
-
-		for _, o := range ops {
-			switch o.Status {
-			case op.Success:
-				success++
-			case op.Fail:
-				fail++
-			case op.Skip:
-				skip++
-			default:
-				unknown++
-			}
-		}
-
-		_, err := fmt.Fprint(t, formatReportLine(
-			prod,
-			strconv.Itoa(success),
-			strconv.Itoa(fail),
-			strconv.Itoa(skip),
-			strconv.Itoa(unknown),
-			strconv.Itoa(len(ops))))
-		if err != nil {
-			return err
-		}
-	}
-
-	err = t.Flush()
-	if err != nil {
-		return err
-	}
-
-	return err
-}
-
-func formatReportLine(cells ...string) string {
-	format := ""
-
-	// The coercion from the argument of type []string to type []interface is required for the later
-	// call to fmt.Sprintf, in which variadic arguments must be of type any/interface{}.
-	strValues := make([]interface{}, len(cells))
-	for i, cell := range cells {
-		format += "%s\t"
-		strValues[i] = cell
-	}
-
-	format += "\n"
-
-	return fmt.Sprintf(format, strValues...)
+// ResultsDest is provided for read-only access to the destination where the agent writes its results. The particular
+// destination is determined by the agent while running; it is unexported to avoid accidental overwrite by external
+// packages. However, its value is useful to know for downstream user interaction, so this method intends to provide
+// that ability.
+func (a *Agent) ResultsDest() string {
```

Comment:
I waffled on just exporting the destination field or adding a public getter method. The doc comment describes why I landed on this, but I would be open to other opinions here for sure.